src/clj/ataru/organization_service/session_organizations.clj

```diff
@@ -28,12 +50,16 @@
                           when-ordinary-user-fn
                           when-superuser-fn]
   {:pre [(right-seq? rights)]}
-  (let [organizations     (select-organizations-for-rights session rights)
-        organization-oids (set (map :oid organizations))]
+  (let [organizations         (select-organizations-for-rights organization-service session rights)
+        organization-oids     (set (map :oid organizations))
+        selected-organization (:selected-organization session)]
     (cond
       (empty? organizations)
       (when-no-orgs-fn)
 
+      (some? selected-organization)
```

Comment:
`select-organizations-for-rights` muutoksen jälkeen tämä haara palauttaa aina saman tuloksen kuin `:else` haara.